packages/plasma-new-hope/src/components/Editable/Editable.tsx

```diff
@@ -0,0 +1,176 @@
+                        navigator.clipboard.writeText(text);
+                    } catch {
+                        inputRef.current.textContent = text;
+                    }
+                }
+
+                if (onPaste) {
+                    onPaste(e as ClipboardEvent<HTMLInputElement>);
+                }
+            };
+
+            const handleBlur: FocusEventHandler<HTMLDivElement> = (e) => {
+                setIsEditing(false);
+
+                clearSelection();
+
+                if (onBlur) {
+                    onBlur(e as FocusEvent<HTMLInputElement>);
+                }
+            };
+
+            const handleFocus: FocusEventHandler = () => {
+                setIsEditing(true);
+
+                inputRef.current && selectText(inputRef.current);
+            };
+
+            const handleEditClick = () => {
+                inputRef.current?.focus();
+            };
+
+            const handleKeyDown: KeyboardEventHandler = (e) => {
+                if ([constants.keyCodes.Enter, constants.keyCodes.Escape].includes(e.keyCode)) {
+                    inputRef.current?.blur();
+                }
+            };
+
+            const handleChange: FormEventHandler = (e) => {
+                if (!inputRef.current) {
+                    return;
+                }
+
+                const contentLength = inputRef.current.textContent?.length || 0;
+
+                if (!maxLength || contentLength <= maxLength) {
+                    prevValueRef.current = inputRef.current.textContent || '';
+
+                    onChange && onChange(e as ChangeEvent<HTMLInputElement>);
+                } else {
+                    inputRef.current.textContent = prevValueRef.current;
+                }
+            };
+
+            useEffect(() => {
+                if (!inputRef.current) {
+                    return;
+                }
+
+                if (typeof value === 'undefined') {
+                    return;
+                }
+
+                if (value === inputRef.current.textContent) {
+                    return;
+                }
+
+                inputRef.current.textContent = value;
+            }, [value, textComponent]);
+
+            return (
+                <Root {...props}>
+                    <Component
+                        ref={innerRef}
+                        className={classes.editableTextBox}
+                        role="textbox"
+                        spellCheck={spellCheck}
+                        contentEditable
+                        onPaste={handlePaste}
+                        onInput={handleChange}
+                        onBlur={handleBlur}
+                        onFocus={handleFocus}
+                        onKeyDown={handleKeyDown}
+                        placeholder={placeholder}
+                        aria-label={placeholder}
+                    />
+                    <input type="hidden" value={value ?? String(inputRef.current?.textContent)} name={name} />
```

Comment:
Напомни, пожалуйста, у нас нет такого, что мы всегда теги оборачиваем в `Styled`, даже если не накидываем сверху стили? Я просто встречал такое у нас во многих компонентах.

```suggestion
                    <StyledInput type="hidden" value={value ?? String(inputRef.current?.textContent)} name={name} />
```